### ISSUE TITLE
eastwest-gen fails to get ready in some tests

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#27916]
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/27895/integ-security-k8s-tests_istio/21642

from https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/27895/integ-security-k8s-tests_istio/21642/artifacts/security-sds-ingress-4bf5f213ec/_suite_context/istio-state-553744279/cluster-0/istio-eastwestgateway-69c756d9cc-k928l_pod-state.yaml, 
```
    {
        "name": "istio-proxy",
        "state": {
          "waiting": {
            "reason": "ContainerCreating"
          }
        },
        "lastState": {

        },
        "image": "localhost:5000/proxyv2:istio-testing",
        "started": false
      }
```

which I think means it didn't even start - some k8s level issue?

Another one - https://prow.istio.io/view/gs/istio-prow/logs/integ-k8s-115_istio_postsubmit/1979

cc @stevenctl

**Comment by stevenctl** [comment]
https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/28296/integ-security-k8s-tests_istio/22996

In #28296 we use `istioctl install` instead of `istioctl generate | kubectl apply` and we're seeing the gateway never ready due to not getting config from pilot. No errors in istiod logs, getting this in eastwest logs:
```
2020-10-28T19:54:01.443443Z	error	xdsproxy	failed to connect to upstream istiod.istio-system.svc:15012: context deadline exceeded
2020-10-28T19:54:01.443811Z	warning	envoy config	StreamAggregatedResources gRPC config stream closed: 2, context deadline exceeded
```

**Comment by stevenctl** [comment]
I believe this is fixed as of #28657  – we _should_ be waiting for Ready state with `istioctl install`.
